File: business/src/main/kotlin/com/isystk/sample/common/dto/CommaSeparatedString.java
package com.isystk.sample.common.dto;

import java.io.Serializable;
import org.apache.commons.lang3.StringUtils;
import org.seasar.doma.Domain;

@Domain(valueType = String.class, factoryMethod = "of")
public class CommaSeparatedString implements Serializable {

  private static final long serialVersionUID = -6864852815920199569L;

  String data;

  public CommaSeparatedString() {
  }

  /**
   * ファクトリメソッド
   *
   * @param data
   * @return
   */
  public static CommaSeparatedString of(String data) {
    CommaSeparatedString css = new CommaSeparatedString();
    css.data = StringUtils.join(data, ",");
    return css;
  }

  // ResultSet.getBytes(int)で取得された値がこのコンストラクタで設定される
  CommaSeparatedString(String data) {
    this.data = StringUtils.join(data, ",");
  }

  // PreparedStatement.setBytes(int, bytes)へ設定する値がこのメソッドから取得される
  String getValue() {
    return this.data;
  }

  public String getData() {
    return this.data;
  }
}

File: business/src/main/kotlin/com/isystk/sample/common/dto/CommaSeparatedString.kt
package com.isystk.sample.common.dto

import org.apache.commons.lang3.StringUtils
import org.seasar.doma.Domain
import java.io.Serializable

@Domain(valueType = String::class, factoryMethod = "of")
class CommaSeparatedString : Serializable {
    // PreparedStatement.setBytes(int, bytes)へ設定する値がこのメソッドから取得される
    var value: String? = null
        get() = field

    constructor() {}

    // ResultSet.getBytes(int)で取得された値がこのコンストラクタで設定される
    internal constructor(data: String) {
        value = StringUtils.join(data, ",")
    }

    companion object {
        private const val serialVersionUID = -6864852815920199569L

        /**
         * ファクトリメソッド
         *
         * @param data
         * @return
         */
        @JvmStatic
        fun of(data: String): CommaSeparatedString {
            val css = CommaSeparatedString()
            css.value = StringUtils.join(data, ",")
            return css
        }
    }
}